luma/core/device.py

```diff
@@ -168,6 +173,62 @@ def display(self, image):
         :param image: Image to display.
         :type image: PIL.Image.Image
         """
-        assert(image.size == self.size)
+        assert image.size == self.size
 
         self.image = self.preprocess(image).copy()
+
+
+class fb(device):
+    def __init__(self, display=None, rotate=0, **kwargs):
+        super(fb, self).__init__(serial_interface=noop())
+        self.id = self.__get_display_id(display)
+        (width, height) = self.__config("virtual_size")
+        self.bits_per_pixel = next(self.__config("bits_per_pixel"))
+        assert self.bits_per_pixel in [24, 32]
+        self.capabilities(width, height, rotate, mode="RGB")
+
+    def __get_display_id(self, display):
+        """
+        Allow for :1 or /dev/fb1, etc
+        """
+        if display is None:
+            display = os.environ.get("DISPLAY")
+
+        if display.startswith(":"):
+            return int(display[1:])
+
+        if display.startswith("/dev/fb"):
+            return int(display[7:])
+
+        raise luma.core.error.DeviceNotFoundError(
+            "Invalid/unsupported framebuffer: {}".format(display)
+        )
+
+    def __config(self, section):
+        path = "/sys/class/graphics/fb{0}/{1}".format(self.id, section)
```

Comment:
can you move this string to a class var/constant?

Reply:
I would prefer to keep it localized to the method. There are no other usages in the class of the string, and it is specific to the config, so keeping it here seems to be the right thing to ensure it is encapsulated.

Reply:
it's used in the tests multiple times, that's why I suggested it. Up to you.